docs/src/main/tut/testing.md

```diff
@@ -0,0 +1,157 @@
+---
+menu: main
+weight: 350
+title: Testing
+---
+
+## Introduction
+
+This document will show a simple Swagger Doc, its implementation as an `org.http4s.HttpService`, and then
+walk through the results of applying inputs, i.e. `org.http4s.Request`, to the Service, i.e. `org.http4s.HttpService`.
+
+After reading this doc, the reader should feel comfortable writing a unit test using his/her favorite Scala testing library.
+
+## Example
+
+Given the following service, which we can define through the following [Swagger](https://swagger.io/) contract:
```

Comment:
I like the swagger use, but is it really relevant to testing?  We're not generating a client off this. :thinking:

Reply:
I added a Swagger doc for a few reasons:

1. habit - at work, before developing a new microservice, we draw up a contract, i.e. Swagger doc, first 
2. I thought it would be helpful to document the service with its contract before implementing it.

Please let me know if you don't think it adds any value. I'm not going to fight to keep it - just want the docs to be clear.

Reply:
I think that approach to developing services has some merit.  It just feels a bit like a distraction from the tut, to me.  I'm mildly in favor of its removal, but if other people like it here, I also won't fight it.

Reply:
I'm not against keeping it, but I have to say I wouldn't read it when reading the tut. Yaml is nice but is still noisy and I only really read it via some nice UI.